### PR TITLE
chore: fix wrong value in documentation for vdc attributes

### PR DESCRIPTION
**Pull request by dmicheneau**
If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### How has this code been tested

```
go test -v ./internal/testsacc/ -count=1 -timeout 30m -run TestVDCDataSource
=== RUN   TestVDCDataSource
--- PASS: TestVDCDataSource (66.26s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc  66.625s
```

```
go test -v ./internal/testsacc/ -count=1 -timeout 30m -run TestAccVDCResource
=== RUN   TestAccVDCResource
--- PASS: TestAccVDCResource (122.82s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc  123.141s
```

### CODE REVIEW

File: internal/testsacc/vdc_datasource_test.go [hunk 1]
```diff
@@ -2,44 +2,77 @@
...
+				Create: testsacc.TFConfig{
+					TFConfig: `
+					data "cloudavenue_vdc" "example" {
+						name = cloudavenue_vdc.example.name
+					}`,
+					// TFConfig: testsacc.GenerateFromTemplate("cloudavenue_vdc",`
+					// data "cloudavenue_vdc" "example" {
+					// 	name = {{ get . "name" }}
+					// }`,
+					// Checks: NewVDCResourceTest().Tests(ctx)["example"](ctx, resourceName).GenerateCheckWithCommonChecks(),
```
Comment (azrod) [inline internal/testsacc/vdc_datasource_test.go:52]:
```suggestion
					Checks: NewVDCResourceTest().Tests(ctx)["example"](ctx, resourceName).GenerateCheckWithCommonChecks(),
```

Use this for using the same test in resource with resourceName in Datasource

File: internal/testsacc/vdc_datasource_test.go [hunk 2]
```diff
@@ -2,44 +2,77 @@
...
+				// ! Create testing
+				Create: testsacc.TFConfig{
+					TFConfig: `
+					data "cloudavenue_vdc" "example" {
+						name = cloudavenue_vdc.example.name
+					}`,
+					// TFConfig: testsacc.GenerateFromTemplate("cloudavenue_vdc",`
+					// data "cloudavenue_vdc" "example" {
+					// 	name = {{ get . "name" }}
+					// }`,
```
Comment (azrod) [inline internal/testsacc/vdc_datasource_test.go:51]:
```suggestion
```

File: internal/provider/vdc/vdc_resource.go [hunk 3]
```diff
@@ -111,26 +111,33 @@ func (r *vdcResource) Create(ctx context.Context, req resource.CreateRequest, re
...
 		},
 	}
 
+	// Get the storage profiles
+	var diags diag.Diagnostics
+	storageProfiles, d := plan.GetVDCStorageProfiles(ctx)
+	diags.Append(d...)
+	if d.HasError() {
+		return
+	}
```
Comment (azrod) [inline internal/provider/vdc/vdc_resource.go:134]:
```suggestion
	// Get the storage profiles
	storageProfiles, d := plan.GetVDCStorageProfiles(ctx)
	resp.Diagnostics.Append(d...)
	if resp.Diagnostics.HasError() {
		return
	}
```

File: internal/provider/vdc/vdc_resource.go [hunk 4]
```diff
@@ -111,26 +111,33 @@ func (r *vdcResource) Create(ctx context.Context, req resource.CreateRequest, re
...
 	// Iterate over the storage profiles and add them to the body.
-	for _, storageProfile := range plan.VDCStorageProfiles {
+	for _, storageProfile := range storageProfiles {
 		body.Vdc.VdcStorageProfiles = append(body.Vdc.VdcStorageProfiles, apiclient.VdcStorageProfilesV2{
-			Class:    storageProfile.Class.ValueString(),
-			Limit:    int32(storageProfile.Limit.ValueInt64()),
-			Default_: storageProfile.Default.ValueBool(),
+			Class:    storageProfile.Class.Get(),
+			Limit:    int32(storageProfile.Limit.Get()),
```
Comment (azrod) [inline internal/provider/vdc/vdc_resource.go:139]:
```suggestion
			Limit:   storageProfile.Limit.GetInt32(),
```

File: internal/provider/vdc/vdc_resource.go [hunk 5]
```diff
@@ -270,40 +278,40 @@ func (r *vdcResource) Read(ctx context.Context, req resource.ReadRequest, resp *
...
-		VDCBillingModel:        types.StringValue(vdc.Vdc.VdcBillingModel),
-		VcpuInMhz2:             types.Float64Value(vdc.Vdc.VcpuInMhz2),
-		CPUAllocated:           types.Float64Value(vdc.Vdc.CpuAllocated),
-		MemoryAllocated:        types.Float64Value(vdc.Vdc.MemoryAllocated),
-		VDCStorageBillingModel: types.StringValue(vdc.Vdc.VdcStorageBillingModel),
-		VDCStorageProfiles:     profiles,
+	state.ID.Set(ID)
+	state.Name.Set(vdc.Vdc.Name)
+	state.Description.Set(vdc.Vdc.Description)
+	state.VDCGroup.Set(vdc.VdcGroup)
```
Comment (azrod) [inline internal/provider/vdc/vdc_resource.go:302]:
```suggestion
	state.VDCGroup.Set(vdc.VdcGroup) // Now due to deprecated field use value in state
```

File: internal/provider/vdc/vdc_resource.go [hunk 6]
```diff
@@ -367,28 +375,36 @@ func (r *vdcResource) Update(ctx context.Context, req resource.UpdateRequest, re
...
+		body.VdcGroup = state.VDCGroup.Get()
+	}
+
+	// Get the storage profiles
+	var diags diag.Diagnostics
+	storageProfiles, d := plan.GetVDCStorageProfiles(ctx)
+	diags.Append(d...)
+	if d.HasError() {
+		return
 	}
```
Comment (azrod) [inline internal/provider/vdc/vdc_resource.go:400]:
```suggestion
	// Get the storage profiles
	storageProfiles, d := plan.GetVDCStorageProfiles(ctx)
	resp.Diagnostics.Append(d...)
	if resp.Diagnostics.HasError() {
		return
	}
```

File: internal/provider/vdc/vdc_resource.go [hunk 7]
```diff
@@ -367,28 +375,36 @@ func (r *vdcResource) Update(ctx context.Context, req resource.UpdateRequest, re
...
 
 	// Iterate over the storage profiles and add them to the body.
-	for _, storageProfile := range plan.VDCStorageProfiles {
+	for _, storageProfile := range storageProfiles {
 		body.Vdc.VdcStorageProfiles = append(body.Vdc.VdcStorageProfiles, apiclient.VdcStorageProfilesV2{
-			Class:    storageProfile.Class.ValueString(),
-			Limit:    int32(storageProfile.Limit.ValueInt64()),
-			Default_: storageProfile.Default.ValueBool(),
+			Class:    storageProfile.Class.Get(),
+			Limit:    int32(storageProfile.Limit.Get()),
```
Comment (azrod) [inline internal/provider/vdc/vdc_resource.go:406]:
```suggestion
			Limit:   storageProfile.Limit.GetInt32(),
```

**Review by azrod — CHANGES_REQUESTED** [review]

File: internal/testsacc/vdc_datasource_test.go (hunk 1 above)
Comment (dmicheneau) [reply internal/testsacc/vdc_datasource_test.go:52]: unable to use due to a problem to get value from template in dependencies
```
 TF_ACC=1 go test  -v ./internal/testsacc/ -count=1 -timeout 30m -run TestVDCDataSource
=== RUN   TestVDCDataSource
    vdc_datasource_test.go:71: Step 1/1 error: Check failed: 2 errors occurred:
                * Check 9/13 error: data.cloudavenue_vdc.example: Attribute 'name' expected "QBWc9h64pGSvazlv", got "C3DZjzXsX2GaIyR0"
                * Check 10/13 error: data.cloudavenue_vdc.example: Attribute 'description' expected "Etiamne flete hi intuerer.", got "Pro conexos meo."
        
--- FAIL: TestVDCDataSource (57.62s)
FAIL
FAIL    github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc  57.988s
FAIL
```

**Review by azrod — APPROVED** [review]

**Pull request merged**